t/api/08-jobtemplates.t

```diff
@@ -631,6 +631,230 @@ $get = $t->get_ok("/api/v1/experimental/job_templates_scheduling/1001")->status_
+push @{$yaml->{foo}{architectures}{i586}{'opensuse-13.1-DVD-i586'}}, $test_suite;
+$t->post_ok(
+    '/api/v1/experimental/job_templates_scheduling',
+    form => {
+        template => YAML::XS::Dump($yaml)}
+)->status_is(200)->json_is(
+    '' => {
+        id => $job_group_id3,
+    },
+    "Added test $test_suite via the template"
+);
+$get = $t->get_ok("/api/v1/experimental/job_templates_scheduling/$job_group_id3");
+is_deeply(YAML::XS::Load($get->tx->res->body), $yaml, "Added test $test_suite to the database")
+  || diag explain($get->tx->res->body, YAML::XS::Dump($yaml));
+
+# Remove test from group according to YAML template
+$test_suite = pop @{$yaml->{foo}{architectures}{i586}{'opensuse-13.1-DVD-i586'}};
+$t->post_ok(
+    '/api/v1/experimental/job_templates_scheduling',
+    form => {
+        template => YAML::XS::Dump($yaml)}
+)->status_is(200)->json_is(
+    '' => {
+        id => $job_group_id3,
+    },
+    'Dropped test by removing it from the template'
+);
+$get = $t->get_ok("/api/v1/experimental/job_templates_scheduling/$job_group_id3");
+is_deeply(YAML::XS::Load($get->tx->res->body), $yaml, "Dropped test $test_suite from the database")
+  || diag explain($get->tx->res->body, YAML::XS::Dump($yaml));
+
+$test_suite = pop @{$yaml->{foo}{architectures}{i586}{'opensuse-13.1-DVD-i586'}};
+$t->post_ok(
+    '/api/v1/experimental/job_templates_scheduling',
+    form => {
+        template => YAML::XS::Dump($yaml)}
+)->status_is(200)->json_is(
+    '' => {
+        id => $job_group_id3,
+    },
+    'Dropped test by removing it from the template'
+);
+$get = $t->get_ok("/api/v1/experimental/job_templates_scheduling/$job_group_id3");
+is_deeply(YAML::XS::Load($get->tx->res->body), $yaml, "Dropped test $test_suite from the database")
+  || diag explain($get->tx->res->body, YAML::XS::Dump($yaml));
+
+# Modify test attributes in group according to YAML template
+$yaml->{foo}{architectures}{i586}{'opensuse-13.1-DVD-i586'}
+  = [{'foobar' => {priority => 11, machine => '32bit'}}, 'spam', 'eggs',];
+# Use per-arch default priority which deviates from the group default_priority
+$yaml->{foo}{defaults}{i586}{'priority'} = 70;
+$t->post_ok(
+    '/api/v1/experimental/job_templates_scheduling',
+    form => {
+        template => YAML::XS::Dump($yaml)}
+)->status_is(200)->json_is(
+    '' => {
+        id => $job_group_id3,
+    },
+    'Test suite was updated'
+);
+# Prepare expected result
+$yaml->{foo}{architectures}{i586}{'opensuse-13.1-DVD-i586'}
+  = [{'foobar' => {priority => 11, machine => '32bit'}}, {'spam' => {priority => 70}}, {'eggs' => {priority => 70}},];
+# Result *should* in fact be 70, but we get the default_priority
+$yaml->{foo}{defaults}{i586}{'priority'} = 50;
+$get = $t->get_ok("/api/v1/experimental/job_templates_scheduling/$job_group_id3");
+is_deeply(YAML::XS::Load($get->tx->res->body), $yaml, 'Modified test suite should be reflected in the database')
+  || diag explain($get->tx->res->body, YAML::XS::Dump($yaml));
```

Comment:
I'm wondering why you pass a 2nd parameter to the `explain` function. I've used the Perl debugger to find out what this does because it isn't documented and it seems like it would just print the 2nd parameter as well. But that's not very useful because the contents of that variable are present in the test code itself. So I'd just remove the 2nd argument here (and also in the other occasions).

Reply:
That's to highlight if the change isn't obvious, since is_deeply won't show any context. Although I don't have a strong opinion on this...